src/hangar/cli/cli.py

```diff
@@ -535,136 +536,172 @@ def server(overwrite, ip, port, timeout):
+@click.option('-f', '--format', 'format_', required=False, help='File format of output file')
 @click.option('--plugin', required=False, help='override auto-inferred plugin')
 @pass_repo
-def export_data(repo: Repository, startpoint, arrayset, out, sample, format_, plugin):
-    """export ARRAYSET sample data as it existed a STARTPOINT to some format and path.
+@click.pass_context
+def export_data(ctx, repo: Repository, arrayset, outdir, startpoint, sample, format_, plugin):
+    """Export ARRAYSET sample data as it existed a STARTPOINT to some format and path.
+
+    Specifying which sample to be exported is possible by using the switch
+    `--sample` (without this, all the samples in the given arrayset will be
+    exported). Since hangar supports both int and str datatype for the sample
+    name, specifying that while mentioning the sample name might be necessary
+    at times. It is possible to do that by separating the name and type by a
+    colon.
+
+    Example:
+
+       1. if the sample name is string of numeric 10 - ``10:str`` or ``10``
+
+       2. if the sample name is ``sample1`` - ``sample1:str1`` or ``sample1``
+
+       3. if the sample name is an int, let say 10 - ``10:int``
     """
+    # TODO: ignore warning through env variable
     from hangar.records.commiting import expand_short_commit_digest
-    from hangar.records.heads import get_branch_head_commit
-    from hangar.cli.io import imsave
+    from hangar.records.heads import get_branch_head_commit, get_staging_branch_head
+    from hangar import external
+
+    kwargs = parse_custom_arguments(ctx.args)
 
     if startpoint in repo.list_branches():
         base_commit = get_branch_head_commit(repo._env.branchenv, startpoint)
-    else:
+    elif startpoint:
         base_commit = expand_short_commit_digest(repo._env.refenv, startpoint)
+    else:
+        branch_name = get_staging_branch_head(repo._env.branchenv)
+        base_commit = get_branch_head_commit(repo._env.branchenv, branch_name)
+    co = repo.checkout(commit=base_commit)
 
     try:
-        co = repo.checkout(write=False, commit=base_commit)
-        arrayset = co.arraysets[arrayset]
+        aset = co.arraysets.get(arrayset)
         if sample:
+            sample, stype = sample.split(':') if ':' in sample else (sample, '')
+            sample = int(sample) if stype == 'int' else str(sample)
             sampleNames = [sample]
         else:
-            sampleNames = list(arrayset.keys())
+            sampleNames = list(aset.keys())
 
-        if format_:
-            format_ = format_.lstrip('.')
-        outP = os.path.expanduser(os.path.normpath(out))
+        extension = format_.lstrip('.') if format_ else None
+        outdir = os.path.normpath(os.path.expanduser(outdir))
+        if not os.path.isdir(outdir):
+            click.echo(f"Directory {outdir} does not exist")
+            return
```

Comment:
you don't need to do any of this. I'm pushing up a change that shows the fix implemented in the `import_data` method, but briefly. 

- use the `click.Path` type in the option specification. for this method this would make click check the path exists, is a directory, is not a file, is writeable by the user, and is fully resolved (including expanding symlinks / hardlinks - which this code doesn't catch)
```suggestion
@click.option('-o', '--out', 'outdir',
                required=True,
                type=click.Path(exists=True, dir_okay=True, file_okay=False, writeable=True, resolve_path=True))
```

Also, we should be using `click.ClickException` instead of "echo and return". This will still except immediatly, but won't print the traceback by default, and makes it possible during testing to ensure that this particular case raised an error. I don't know if the tests cover this case now, but please make sure they do. See my changes in `import_data` for an example.

Reply:
Done, thanks for pointing it out